ivantiitsm_connector.py

```diff
@@ -146,8 +147,11 @@ def _suds_to_dict(self, sud_obj):
             return sud_obj.strftime(consts.HEAT_TIME_FORMAT)
 
         # Checking for NaN
-        elif sud_obj != sud_obj:
-            return None
+        try:
+            if math.isnan(float(sud_obj)):
+                return None
+        except:
+            self.debug_print('{} is not NaN'.format(sud_obj))
```

Comment:
```suggestion
            self.debug_print('{} is not a numeric value'.format(sud_obj))
```
or maybe just debug print the exception itself?

NaN stands for "not a number", so the log currently says "not not a number". iinm if we enter this except block it's because `sud_obj` wasn't a number and thus the `float(...)` call failed?

Reply:
Good catch! Yep, try-catch is being used here to handle the case where `sud_obj` is not a number.